### ISSUE TITLE
msgctxt is incorrect

### ISSUE THREAD

**Issue by BJury** [OrchardCoreContrib/OrchardCoreContrib.PoExtractor#30]
The msgctxt includes the whole path which doesn't work with OC's localisation features.
eg

for a string in 'c:\projects\OCProject\XX.OrchardCore.Modul\Views\home.cshtml'

```msgctxt "c:.projects.OCProject.XX.OrchardCore.Module.Views.Home.Get"```

should be 

```msgctxt "XX.OrchardCore.Module.Views.Home.Get"```

**Comment by hishamco** [comment]
Could you please try this again on the latest bits?

**Comment by Patrickkk** [comment]
With the latest i don't have this issue, but i seem to have the issue that OrchardCore.Localization.Core seems to expect a message context without the project name.

For exampel using the extractor for a project with the project name "POLocalization.Context" it will output "POLocalization.Context.Views.Home.Index" as a context, but it won't be translated because it is expecting only "Views.Home.Index"
```msgctxt "Views.Home.Index"
msgid "Translated"
msgstr "Vertaald"

msgctxt "POLocalization.Context.Views.Home.Index"
msgid "Not translated"
msgstr "Niet vertaald"```

Now this might be fine for orchard, but it would be nice if its possible to remove this via a flag so it can be used outside of orchard

**Comment by Patrickkk** [comment]
[POLocalization.Context.zip](https://github.com/lukaskabrt/PoExtractor/files/5551881/POLocalization.Context.zip)
Example project showcasing what I mean

**Comment by hishamco** [comment]
I will have a look ...

**Comment by hishamco** [comment]
@Patrickkk I checked the sample and I realize what you mean, removing the project name / namespace from the context is worked for your sample, but what if we have `Views.Home.Index` in two projects?!! Hope you know what I mean, I will do few test, probably may I need to modify something in OC.Localization.Core

**Comment by Patrickkk** [comment]
👍  thanks for cheaking. Yeah that makes sense. The more complete context would seem better to prevent 'collisions' but this won't work with OC.Localization.Core indeed. Should I make a seperate github issue there?

**Comment by hishamco** [comment]
Let me complete few unit tests, then we can file an issue if it's required

Thanks

**Comment by hishamco** [comment]
Is this resolved?

**Comment by Patrickkk** [comment]
Were there any changes related to this? Last I checked the issues was still the same.
Just checked the example project from earlier with the latest nuget for OrchardCore.Localization.Core and things are still the same. But it seems to me its an issue with "OrchardCore.Localization.Core" not working with the full msgctxt including the project file.

**Comment by hishamco** [comment]
Could you please file the issue in OC explaining the steps to reproduce

Cc me

**Comment by hishamco** [comment]
@BJury @Patrickkk is this issue still exist?

**Comment by Skrypt** [comment]
I'm using older bits from the previous project but when I'm using `extractpo` command line I get msgctx with \\ instead of `.` everywhere. Though `extractpo-oc` doesn't do that. I will try your new release here and see if it does the same.

**Comment by hishamco** [comment]
Thanks @Skrypt for pointing to that, I might test this before release the NuGet package, but the question how everything works fine with @agriffard in Crowding and `OC.Translations` repo

**Comment by Skrypt** [comment]
This issue happens when I am using a relative path.

```
extractpo-oc "..\..\..\..\src" "..\..\..\..\tmp\translations"
```

This will create bad entries in the .pot files by adding the full folder path.
Using fixed absolute path works.

I reverted back to version 0.3.1-beta3-71077 and the issue is there too.
Will now test back with latest 0.5.0-rc2-16220 and report what find.

**Comment by Skrypt** [comment]
Here with 0.5.0-rc2-16220 still getting an issue though.

```
#: OrchardCore.Themes.Poseidon\Views\OrchardCore.Diagnostics\Diagnostics\Unauthorized.cshtml:0
msgctxt "OrchardCore.Themes.Poseidon\\Views\\OrchardCore.Diagnostics\\Diagnostics\\Unauthorized"
msgid "You are not authorized to view this content."
msgstr ""
```

You can see on the msgctxt where it adds `\\` instead of `.`

**Comment by hishamco** [comment]
After a quick try what i see is:

**Absolute Path**

```
#: Admin\Views\Layout.cshtml:46
#. <a id="btn-darkmode" class="me-3" data-bs-toggle="tooltip" data-bs-placement="bottom" title="@(darkMode ? T["Light mode"] : T["Dark mode"])" role="button">
#: Admin\Views\Layout.cshtml:50
#. <span class="d-none">@T["Light mode"]</span>
msgctxt "Admin.Views.Layout"
msgid "Light mode"
msgstr ""
```

**Relative Path**

```
#: D:\Git\Repositories\OrchardCoreContrib.PoExtractor\OrchardCore.Themes\TheAdmin\Views\Layout.cshtml:46
#. <a id="btn-darkmode" class="me-3" data-bs-toggle="tooltip" data-bs-placement="bottom" title="@(darkMode ? T["Light mode"] : T["Dark mode"])" role="button">
#: D:\Git\Repositories\OrchardCoreContrib.PoExtractor\OrchardCore.Themes\TheAdmin\Views\Layout.cshtml:50
#. <span class="d-none">@T["Light mode"]</span>
msgctxt "D:.Git.Repositories.OrchardCoreContrib.PoExtractor.OrchardCore.Themes.TheAdmin.Views.Layout"
msgid "Light mode"
msgstr ""
```

Seems the issue is using relative path includes the full path which is the thing need to be fixed

**Comment by hishamco** [comment]
Fixed in [1631c2f362c91fc8fdc66d8ac2e7678649f12014](https://github.com/OrchardCoreContrib/OrchardCoreContrib.PoExtractor/commit/1631c2f362c91fc8fdc66d8ac2e7678649f12014)
